clang/labs/lab1/shishkina_valeriya/PrintClasses.cpp

```diff
@@ -0,0 +1,56 @@
+#include "clang/AST/ASTConsumer.h"
+#include "clang/AST/RecursiveASTVisitor.h"
+#include "clang/Frontend/CompilerInstance.h"
+#include "clang/Frontend/FrontendPluginRegistry.h"
+
+class PrintClassesVisitor : public clang::RecursiveASTVisitor<PrintClassesVisitor> {
+public:
+    explicit PrintClassesVisitor(clang::ASTContext *Context) : Context(Context) {}
+
+    bool VisitCXXRecordDecl(clang::CXXRecordDecl *Res) {
+    llvm::outs() << Res->getNameAsString() << "\n";
+   
+    for (clang::Decl *decl : Res->decls()) {
+        if (auto *field = clang::dyn_cast<clang::FieldDecl>(decl)) {
+            llvm::outs() << "  |_ " << field->getNameAsString() << "\n";
+        } else if (auto *staticField = clang::dyn_cast<clang::VarDecl>(decl)) {
+            if (staticField->isStaticDataMember()) {
+                llvm::outs() << "  |_ " << staticField->getNameAsString() << "\n";
+            }
+        }
+    }
+    return true;
+}
+
+private:
+    clang::ASTContext *Context;
+};
+
+class PrintClassesConsumer : public clang::ASTConsumer {
+public:
+    explicit PrintClassesConsumer(clang::ASTContext *Context) : Visitor(Context) {}
+
+    void HandleTranslationUnit(clang::ASTContext &Context) override {
+        Visitor.TraverseDecl(Context.getTranslationUnitDecl());
+    }
+
+private:
+    PrintClassesVisitor Visitor;
+};
+
+class PrintClassesPlugin : public clang::PluginASTAction {
+public:
+  std::unique_ptr<clang::ASTConsumer> CreateASTConsumer (
+    clang::CompilerInstance &Compiler, llvm::StringRef InFile) override {
+      return std::make_unique<PrintClassesConsumer>(&Compiler.getASTContext());
+    }
+
+protected:
+  bool ParseArgs(const clang::CompilerInstance &Compiler, 
+                 const std::vector<std::string> &args) override {
```

Comment:
Could you please fix clang-tidy warnings if applicable: https://github.com/NN-complr-tech/llvm-nnsu-2024/actions/runs/8310037939/job/22742071140?pr=18#step:6:489